File: clojure/src/ch1/ex120.clj
;; (load-file "filename.clj")

(defn gcd [a b]
  (cond (= b 0) (do
                  (println "returning" a)
                  a)
        :else (do
                (println "remainder" a b)
                (gcd b (rem a b)))))


(require
 'clojure.test)

(clojure.test/is (= (gcd 206 40) 2))
(clojure.test/is (= (gcd 24 12) 12))
(clojure.test/is (= (gcd 60 45) 15))

(println "done")

File: elixir/ch1/ex120.ex
# cd("c:/Users/Yohan/IdeaProjects/sicp/elixir/ch1")
# Load with import_file("filename")
# Run with Module.func(args)
defmodule Ex120 do
  @moduledoc false

  def gcd(a, 0), do: a
  def gcd(a, b) do
    gcd(b, rem(a, b))
  end

end
